src/rrule.js

```diff
@@ -0,0 +1,1238 @@
+import Weekday from './weekday'
+import dateutil from './dateutil'
+import {
+  range,
+  repeat,
+  pymod,
+  divmod,
+  plb,
+  contains
+} from './helpers'
+
+import {
+  WDAYMASK,
+  M365MASK,
+  M365RANGE,
+  M366MASK,
+  M366RANGE,
+  MDAY365MASK,
+  MDAY366MASK,
+  NMDAY365MASK,
+  NMDAY366MASK
+} from './masks'
+
+import IterResult from './iterresult'
+import CallbackIterResult from './callbackiterresult'
+
+// =============================================================================
+// RRule
+// =============================================================================
+
+/**
+ *
+ * @param {Object?} options - see <http://labix.org/python-dateutil/#head-cf004ee9a75592797e076752b2a889c10f445418>
+ *        The only required option is `freq`, one of RRule.YEARLY, RRule.MONTHLY, ...
+ * @constructor
+ */
+export default class RRule {
+  constructor (options, noCache) {
+    options = options || {}
+    // RFC string
+    this._string = null
+    this._cache = noCache ? null : {
+      all: false,
+      before: [],
+      after: [],
+      between: []
+    }
+
+    // used by toString()
+    this.origOptions = {}
+
+    this.options = {}
```

Comment:
Definitely not technically necessary, and maybe its just a personal thing, but I think it would really help with readability to declare all these class properties (both public and private) above the constructor with some comments as to what they do. I write in typescript and the convention I've seen there is
```
public options;
private _cache;

// ...

constructor() {

// ...
```
Obviously this isn't typescript, but something like
```
/*
 * Options (public)
 * The options argument mostly corresponds to the properties defined for RRULE
 * in the iCalendar RFC. Only freq is required.
 *
 * [Steal the rest of the description from the README]
 */
var options;
```

I realize that library is *currently* lacking annotations like this though, so even without these comments I think this PR is an improvement.

Reply:
Oops! too slow